integration/connection.tf

```diff
@@ -1,37 +1,25 @@
-resource "materialize_connection" "ssh_connection" {
-  name            = "ssh_connection"
-  connection_type = "SSH TUNNEL"
-  ssh_host        = "example.com"
-  ssh_port        = 22
-  ssh_user        = "example"
-}
-
-resource "materialize_connection" "kafka_connection" {
+resource "materialize_connection_kafka" "kafka_connection" {
   name            = "kafka_connection"
-  connection_type = "KAFKA"
   kafka_broker {
     broker = "kafka:9092"
   }
 }
 
-resource "materialize_connection" "schema_registry" {
-  name                          = "schema_registry_connection"
-  connection_type               = "CONFLUENT SCHEMA REGISTRY"
-  confluent_schema_registry_url = "http://schema-registry:8081"
+resource "materialize_connection_confluent_schema_registry" "schema_registry" {
+  name = "schema_registry_connection"
+  url  = "http://schema-registry:8081"
 }
 
-resource "materialize_connection" "example_ssh_connection" {
-  name            = "ssh_example_connection"
-  schema_name     = "public"
-  connection_type = "SSH TUNNEL"
-  ssh_host        = "ssh_host"
-  ssh_user        = "ssh_user"
-  ssh_port        = 22
+resource "materialize_connection_ssh_tunnel" "example_ssh_connection" {
```

Comment:
Just to differentiate from the examples, maybe we should leave "example" out of the integration test resources.